lib/audio_service.dart

```diff
@@ -1701,7 +1701,7 @@ abstract class BackgroundAudioTask {
     if (mediaItem == null) return;
     final queue = AudioServiceBackground.queue ?? [];
     int i = queue.indexOf(mediaItem);
-    if (i == -1) return;
+    if (i == -1 || (i == 0 && offset < 0)) return;
```

Comment:
Good find.

Although perhaps this makes more sense:

```dart
if (i == -1 || offset < -i) return;
```

That said, it might be clearer to add the check down below where it does the incomplete range check (see comment below).

Reply:
Thanks, yeah I agree with you that the solution below is much clearer. Should we go with that one?

Reply:
Sounds good to me. It would be more readable I think to have the range checks together.

Reply:
Agreed! I will change the PR :)